### PR TITLE
Fetch icons with Mempool API

### LINKED ISSUE #542: Use mempool API to retrieve icons dynamically
Eg.
https://liquid.network/api/v1/asset/6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d/icon

### PR DESCRIPTION
**Pull request by Janaka-Steph**
It closes #542 

Please review @tiero

### CODE REVIEW

File: src/components/CurrencyIcon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          `https://liquid.network/api/v1/asset/0e99c1a6da379d1f4151fb9df90449d40d0608f6cb33a5bcbfc8c265f42bab0a/icon`
+        );
+      } else {
+        res = await fetch(`https://liquid.network/api/v1/asset/${assetHash}/icon`);
+        if (!res.ok) {
+          setIcon(null);
+          return;
+        }
+      }
+      const imageBlob = await res.blob();
```
Comment (tiero) [inline src/components/CurrencyIcon/index.tsx:40]: Isn't this base64 serialization expensive?

Why not using a `` component and pass the URL?

File: src/components/CurrencyIcon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d/icon`
+        );
+      } else if (isUsdt(assetHash, network)) {
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/ce091c998b83c78bb71a632313ba3760f1763d9cfcffae02258ffa9865a37bd2/icon`
+        );
+      } else if (isLcad(assetHash, network)) {
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/0e99c1a6da379d1f4151fb9df90449d40d0608f6cb33a5bcbfc8c265f42bab0a/icon`
```
Comment (tiero) [inline src/components/CurrencyIcon/index.tsx:31]: I would have kept featured assets from local storage

File: src/components/CurrencyIcon/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      const imageBlob = await res.blob();
+      reader.readAsDataURL(imageBlob);
+      reader.onloadend = () => {
+        const base64data = reader.result;
+        setIcon(base64data?.toString() ?? null);
+      };
+    })();
+  }, [assetHash, network]);
+
+  return (
```
Comment (tiero) [inline src/components/CurrencyIcon/index.tsx:49]: You can copy this component https://github.com/vulpemventures/marina/blob/9bcffdd7db2b304f06f3a7e52442f585d495fd06/src/presentation/components/assetIcon.tsx

It uses the onError directive and uses the src to retrieve via HTTP

File: src/components/CurrencyIcon/index.tsx (hunk 1 above)
Comment (Janaka-Steph) [reply src/components/CurrencyIcon/index.tsx:40]: Ha yes indeed. I did that for dashboard, to make it work with the AntD Icon component but doesn't apply here.

File: src/components/CurrencyIcon/index.tsx (hunk 2 above)
Comment (tiero) [reply src/components/CurrencyIcon/index.tsx:31]: https://github.com/vulpemventures/marina/blob/9bcffdd7db2b304f06f3a7e52442f585d495fd06/src/application/utils/constants.ts#L13-L51

File: src/components/CurrencyIcon/index.tsx (hunk 3 above)
Comment (Janaka-Steph) [reply src/components/CurrencyIcon/index.tsx:49]: Ok. onError is deprecated tho https://caniuse.com/mdn-html_elements_img_onerror

**Review by tiero — APPROVED** [review]

**Pull request merged**